repo/fsrepo/datastores.go

```diff
@@ -410,3 +412,36 @@ func (c *badgerdsDatastoreConfig) Create(path string) (repo.Datastore, error) {
 
 	return badgerds.NewDatastore(p, &defopts)
 }
+
+// PostgresDatastoreConfig returns a postgres DatastoreConfig from a spec
+func PostgresDatastoreConfig(params map[string]interface{}) (DatastoreConfig, error) {
+	var c postgresDatastoreConfig
+	var ok bool
+	c.path, ok = params["path"].(string)
+	if !ok {
+		log.Fatal("no path")
+	}
+
+	return &c, nil
+}
+
+type postgresDatastoreConfig struct {
+	path string
+}
+
+func (c *postgresDatastoreConfig) DiskSpec() DiskSpec {
+	return map[string]interface{}{
+		"type": "postgresds",
+		"path": c.path,
+	}
+}
+
+func (postgresDatastoreConfig) Create(path string) (repo.Datastore, error) {
+	pg := postgresdb.Options{}
```

Comment:
This should get Host/User/Password/Database from config.

When initializing node with this backend user could pass those using env vars. in [config profile](https://github.com/ipfs/go-ipfs-config/pull/5/files#diff-e72534bab7cc380ab4281a57e79e8186R141) we'd just call `os.Getenv`.

We then could provide a simple script like the one below to avoid storing passwords in shell history
```bash
#!/bin/sh

[check args]

export IPFS_PG_HOST=$1
export IPFS_PG_USER=$2
export IPFS_PG_DATABASE=$3
printf "${IPFS_PG_USER}@${IPFS_PG_HOST} password:"
stty -echo
read IPFS_PG_PASSWORD
stty echo
export IPFS_PG_PASSWORD

ipfs init --profile=postgres
```

Reply:
Thanks for the feedback! Making these changes now. I think we could also use the PGPASSFILE env variable to avoid directly storing the password as an env variable.